plugins/modules/swim_intent.py

```diff
@@ -1362,6 +1379,125 @@ def get_diff_merged(self, config):
+            tag_image_golden = tagging_details.get("tagging")
+
+            image_params = dict(
+                image_id=self.have.get("tagging_image_id"),
+                site_id=self.have.get("site_id"),
+                device_family_identifier=self.have.get("device_family_identifier"),
+                device_role=tagging_details.get("device_role", "ALL").upper()
+            )
+            self.log("Parameters for checking the status of image: {0}".format(str(image_params)), "INFO")
+
+            response = self.dnac._exec(
+                family="software_image_management_swim",
+                function='get_golden_tag_status_of_an_image',
+                op_modifies=True,
+                params=image_params
+            )
+            self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
+
+            response = response.get('response')
+            if response:
+                image_status = response['taggedGolden']
+                if image_status == tag_image_golden:
+                    if tag_image_golden:
+                        self.msg = """The requested image '{0}' has been tagged as golden in the Cisco Catalyst Center and image status
+                                has been verified.""".format(name)
+                        self.log(self.msg, "INFO")
+                    else:
+                        self.msg = """The requested image '{0}' has been un-tagged as golden in the Cisco Catalyst Center and
+                                image status has been verified.""".format(name)
+                        self.log(self.msg, "INFO")
+            else:
+                self.log("""Mismatch between the playbook input for tagging/un-tagging image as golden and the Cisco Catalyst Center indicates that
+                         the tagging/un-tagging task was not executed successfully.""", "INFO")
+
+        if self.want.get("distribution_details"):
+            image_id = self.have.get("distribution_image_id")
+
+            if self.have.get("distribution_device_id"):
+                if self.single_device_distribution:
+                    self.msg = """The requested image with device id '{0}' has been distributed successfully in the Cisco Catalyst Center and
+                             image status has been verified.""".format(self.have.get("distribution_device_id"))
+                    self.log(self.msg, "INFO")
+                else:
+                    self.log("""Mismatch between the playbook input for distributing image to device with id '{0}' and the Cisco Catalyst Center indicates that
+                         the distribution task was not executed successfully.""".format(self.have.get("distribution_device_id")), "INFO")
+            elif self.complete_successful_distribution:
+                self.msg = """The requested image with id '{0}' distributed successfully on all devices in that particular site
```

Comment:
The requested image, with ID '{0}', has been successfully distributed to all devices within the specified site in the Cisco Catalyst Center.".format(image_id)

Can we specify the site?
Can we print the devices (may be IP addresses) ?

Reply:
For now I added the message, and we are printing device ips while executing the process. But here it'll be little complicated so will take once all pending items get done.